### PR TITLE
Keygenerator refactor

### PR DESCRIPTION
**Pull request by iulianpascalau**
- refactored keygenerator app: 
1. added the possibility to output data on the console (through the extra `-console-out` flag)
2. create a single file with all the generated keys (based on type) (through the extra `no-split` flag)

This will also solve issue #1982

### CODE REVIEW

File: cmd/keygenerator/main.go [hunk 1]
```diff
@@ -97,141 +132,338 @@ func main() {
...
+func backupFileIfExists(filename string) {
+	if _, err := os.Stat(filename); err != nil {
+		if os.IsNotExist(err) {
+			return
+		}
+	}
+	//if we reached here the file probably exists, make a timestamped backup
+	_ = os.Rename(filename, filename+"."+fmt.Sprintf("%d", time.Now().Unix()))
 }
+
```
Comment (SebastianMarian) [inline cmd/keygenerator/main.go:374]: Don't forget to remove the commented code

Comment (iulianpascalau) [reply cmd/keygenerator/main.go:374]: done, thanks

File: cmd/keygenerator/main.go [hunk 2]
```diff
@@ -97,141 +132,338 @@ func main() {
...
+	absPath, err := os.Getwd()
 	if err != nil {
-		return err
+		return "", err
 	}
 
-	defer func() {
-		_ = file.Close()
-	}()
+	if numKeys > 1 {
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply cmd/keygenerator/main.go:351]: right, done

File: cmd/keygenerator/main.go [hunk 3]
```diff
@@ -97,141 +132,338 @@ func main() {
...
 			return err
 		}
 
-		return generateTxKey(index, numKeys)
-	default:
-		return fmt.Errorf("unknown key type %s", argsConfig.keyType)
+		data = append(data, string(buf.Bytes()))
 	}
+
+	log.Info(strings.Join(data, ""))
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply cmd/keygenerator/main.go:257]: no, the pem generator automatically writes a `\n` character. Found out through testing.

[the remark this answers is not in the document]
Comment (SebastianMarian) [reply cmd/keygenerator/main.go:257]: ok

**Review by SebastianMarian — DISMISSED** [review]

File: cmd/keygenerator/main.go [hunk 4]
```diff
@@ -97,141 +132,243 @@ func main() {
...
 
-	return core.SaveSkToPemFile(file, pkString, []byte(hex.EncodeToString(sk)))
+func backupFileIfExists(filename string) {
+	if _, err := os.Stat(filename); err != nil {
+		if os.IsNotExist(err) {
+			return
+		}
+	}
+	//if we reached here the file probably exists, make a timestamped backup
+	_ = os.Rename(filename, filename+"."+fmt.Sprintf("%d", time.Now().Unix()))
```
Comment (SebastianMarian) [inline cmd/keygenerator/main.go:373]: Maybe you can change the order here just to keep the extension as .pem to be recognized by wallet when you want to select pem files:
_ = os.Rename(filename, fmt.Sprintf("%d", time.Now().Unix())+"-"+filename)

Comment (iulianpascalau) [reply cmd/keygenerator/main.go]: right, old implementation. Fixed

**Review by SebastianMarian — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by stefanandy — APPROVED** [review]
System test passed

**Pull request merged**